ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -79,6 +77,12 @@ public VcrReplicationManager(VerifiableProperties properties, CloudConfig cloudC
     this.cloudStorageCompactor =
         cloudConfig.cloudBlobCompactionEnabled ? new CloudStorageCompactor(cloudDestination, cloudConfig,
             partitionToPartitionInfo.keySet(), vcrMetrics) : null;
+    // We need a datacenter to replicate from. For DR, we currently deploy VCR nodes on same datacenter as ambry-server.
+    // For serving data from cloud, we must specify the ambry-server datacenter(s) from which to replicate.
+    if (!cloudConfig.cloudIsVcr && cloudConfig.vcrCrossColoReplicationPeerDatacenters.isEmpty()) {
```

Comment:
I don't have much context of cloud but I am trying to understand the config `cloudIsVcr`, so there are two types of nodes related to Cloud. One is VCR, in my impression, it is supposed to replicate blobs from regular ambry server to Azure (correct me if I am wrong). What is the other type of cloud node?

Reply:
cloudIsVcr is used to tell the CloudBlobStore if it is being used by the VCR (true) or by the frontend (live serving).  VcrReplicationManager should never be created when isVcr = false

Reply:
I see. Then, I have some difficulties understanding the `cloudIsVcr` being checked in `VcrReplicationManager`. I suppose it should be VCR already before instantiating the `VcrReplicationManager`.

Reply:
I was under the wrong impression that cloudIsVcr would be true for all cases related to DR, and false otherwise (even though I now realize that comments in CloudBlobStore say so). 
Since thats not the case, I guess the suggestion to always use vcrSourceDatacenters (vcrCrossColoReplicationPeerDatacenters) to get the list of DCs to replicate from makes sense.
I will make the required changes.